libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/LoginActivity.java

```diff
@@ -93,19 +96,31 @@
     private String userHint;
     private String spActivityName;
     private Bundle spActivityExtras;
+    private BlockingQueue<Boolean> authConfigReady;
 
 	@Override
 	protected void onCreate(Bundle savedInstanceState) {
 		super.onCreate(savedInstanceState);
 
-        // Getting login options from intent's extras
+        // Getting login options from intent's extras.
         final LoginOptions loginOptions = LoginOptions.fromBundle(getIntent().getExtras());
 
-        // Protect against screenshots
+        // Protect against screenshots.
         getWindow().setFlags(WindowManager.LayoutParams.FLAG_SECURE,
                 WindowManager.LayoutParams.FLAG_SECURE);
 
-        // Setup content view
+        // Fetches auth config if required.
+        authConfigReady = new ArrayBlockingQueue<>(1);
+        (new AuthConfigTask(this)).execute();
+
+        // Waits for auth config to be fetched before continuing to load the activity.
+        try {
+            authConfigReady.take();
```

Comment:
@wmathurin There's a bit of a problem with this line here. Since it blocks the main thread, it's interfering with the `AsyncTask's MessageQueue` and preventing it from calling `onPostExecute()`. I'm not sure there's a good way to block the main thread and allow `AsyncTask` to post on the main thread without causing a deadlock. The call in `ServerPickerActivity` works beautifully though since it's using the same `AsyncTask` the way it should be - without blocking the main thread, but waiting instead for the callback.

Reply:
Actually, we don't need the BlockingQueue.
You can just do `asyncTask.execute().get()` to wait for the async task to complete.
See: https://stackoverflow.com/a/14046475

Reply:
I tried that already. It didn't work for me. Line of execution continued to the next line. Maybe it's because of `Void` return argument. I'll fiddle around with it a bit.